### PR TITLE
Plug everything

### PR DESCRIPTION
**Pull request by Champii**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/elthariel/blocks/pull/5" target="_blank">ReviewNinja

File: client/block.ls [hunk 1]
```diff
@@ -1,10 +1,7 @@
 require! {
   \../common
-  \./scene
 }
 
-meshNb = 1
 
-common.blocks.Base::deported_ctor = (x, y, z) ->
```
Comment (elthariel) [inline client/block.ls]: Thank you lord.
Thank you

File: client/chunk.ls [hunk 2]
```diff
@@ -1,18 +1,37 @@
...
 
-  ->
+  (@scene) ->
     super!
-    scene!createOrUpdateSelectionOctree();
+    Manager.scene @scene
+    @scene.createOrUpdateSelectionOctree!
 
-  show_near_air: ->
+  show_near_air: (cid) ->
```
Comment (elthariel) [inline client/chunk.ls]: Should not be here -_-

**Comment by elthariel** [comment]
:+1:

**Comment by elthariel** [comment]
I don't know how asio is taking care of its memory, but I don't think you'll be able to easily avoid copy without getting dirty with asio buffer management :-/

My intuition here is that we don't fetch chunks that often (O(1)/second I'd guess). Is it really a big issue to copy them then ?

**Comment by Champii** [comment]
I think it can be easily resolved as we have full management over network abstraction for AMQP. That means that we can implement some kind of circular buffer like the one we had once (or whatever).
(the actual network code is a generic code taken from AMQPCPP exemples...)
I need to dig more on that

**Comment by elthariel** [comment]
> full control :)

circular buffer won't solve this issue. If we need the data to have a
longer lifespan, we'll have to copy it out of the circular buffer

**Pull request merged**